ext/random/tests/engine/user_xoshiro128plusplus.phpt

```diff
@@ -0,0 +1,59 @@
+--TEST--
+Random: Engine: User: Xoshiro128++
+--FILE--
+<?php
+
+/*
+ * Original code by Tim Düsterhus <tim@bastelstu.be>
+ */
+
+final class Xoshiro128PP implements \Random\Engine
+{
+    private function __construct(
+        private int $s0,
+        private int $s1,
+        private int $s2,
+        private int $s3
+    ) {
+    }
+
+    public static function fromNumbers($s0, $s1, $s2, $s3)
+    {
+        return new self($s0, $s1, $s2, $s3);
+    }
+
+    private static function rotl($x, $k)
+    {
+        return (($x << $k) | ($x >> (32 - $k))) & 0xFFFFFFFF;
+    }
+
+    public function generate(): string
+    {
+        $result = (self::rotl(($this->s0 + $this->s3) & 0xFFFFFFFF, 7) + $this->s0) & 0xFFFFFFFF;
+
+        $t = ($this->s1 << 9)  & 0xFFFFFFFF;
+
+        $this->s2 ^= $this->s0;
+        $this->s3 ^= $this->s1;
+        $this->s1 ^= $this->s2;
+        $this->s0 ^= $this->s3;
+
+        $this->s2 ^= $t;
+
+        $this->s3 = self::rotl($this->s3, 11);
+
+        return \pack('L', $result);
```

Comment:
This should use `V` to ensure little-endian encoding on every architecture. `L` is machine dependent.
```suggestion
        return \pack('V', $result);
```

Reply:
fixed fdf1cf486f0388a21b7bea659aa68df19083b54d
Thanks